core/trino-main/src/main/java/io/trino/operator/project/PageProcessor.java

```diff
@@ -118,7 +124,10 @@ public WorkProcessor<Page> createWorkProcessor(ConnectorSession session, DriverY
         }
 
         if (filter.isPresent()) {
-            SelectedPositions selectedPositions = filter.get().filter(session, filter.get().getInputChannels().getInputChannels(page));
+            Page inputPage = filter.get().getInputChannels().getInputChannels(page);
+            long filterStartNanos = System.nanoTime();
```

Comment:
The variable name could be simplified to `start` or `startTime` without loss of clarity.

Reply:
renamed to `start`